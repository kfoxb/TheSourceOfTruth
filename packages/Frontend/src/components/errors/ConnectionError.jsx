import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { database } from 'firebase';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Snackbar from '@material-ui/core/Snackbar';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import colors from '../../constants/colors';

const styles = theme => ({
  close: {
    width: theme.spacing.unit * 4,
    height: theme.spacing.unit * 4,
  },
  root: {
    background: `${colors.darkGrey}`,
  },
});

class SimpleSnackbar extends React.Component {
  state = {
    open: false,
  };

  componentDidMount() {
    const connectedRef = database().ref('.info/connected');
    setTimeout(() => {
      connectedRef.on('value', (snap) => {
        if (snap.val() === true) {
          this.setState({ open: false });
        } else {
          this.setState({ open: true });
        }
      });
    }, 5000);
  }

  handleClick = () => {
    this.setState({ open: true });
  };

  handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    this.setState({ open: false });
  };

  refreshPage = () => {
    window.location.reload();
  }

  render() {
    const { component: Component, classes, ...rest } = this.props;

    return (
      <Fragment>
        <Snackbar
          anchorOrigin={{
            vertical: 'top',
            horizontal: 'center',
          }}
          ContentProps={{
            classes: {
              root: classes.root,
            },
          }}
          open={this.state.open}
          onClose={this.handleClose}
          message={<span id="message-id">CONNECTION LOST: Live updates unavailable</span>}
          action={[
            <Button key="undo" style={{ color: `${colors.green}` }} size="small" onClick={this.refreshPage}>
              REFRESH
            </Button>,
            <IconButton
              key="close"
              aria-label="Close"
              color="inherit"
              className={classes.close}
              onClick={this.handleClose}
            >
              <CloseIcon />
            </IconButton>,
          ]}
          style={{ top: '40px' }}
        />
        <Component {...rest} />
      </Fragment>
    );
  }
}

SimpleSnackbar.propTypes = {
  component: PropTypes.func.isRequired,
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
    close: PropTypes.string.isRequired,
  }).isRequired,
};

export default withStyles(styles)(SimpleSnackbar);
