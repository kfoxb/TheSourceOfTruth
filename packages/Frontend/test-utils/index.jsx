/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    const wrapper = shallow(<Component {...props} />);
    expect(wrapper).toMatchSnapshot();
  };
