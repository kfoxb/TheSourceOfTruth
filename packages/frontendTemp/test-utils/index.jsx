/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const makeShallowCreateWrapper = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    return shallow(<Component {...props} />);
  };

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) => {
  const createWrapper = makeShallowCreateWrapper(Component, defaultProps);
  return (extraProps) => {
    expect(createWrapper(extraProps)).toMatchSnapshot();
  };
};
