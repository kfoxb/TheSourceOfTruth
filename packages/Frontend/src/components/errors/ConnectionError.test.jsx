/* eslint-env jest */
import React from 'react';
import { createAssertWithPropsToMatchSnapshot } from '../../../test-utils';
import ConnectionError from './ConnectionError';

describe('ConnectionError', () => {
  const defaultProps = {
    component: () => (<div />),
    classes: {
      root: '',
      close: '',
    },
  };
  it('should render', () => {
    createAssertWithPropsToMatchSnapshot(ConnectionError, defaultProps)();
  });
});
