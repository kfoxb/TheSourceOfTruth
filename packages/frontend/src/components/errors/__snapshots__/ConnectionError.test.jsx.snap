// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionError should render 1`] = `
<SimpleSnackbar
  classes={
    Object {
      "close": "SimpleSnackbar-close-1",
      "root": "SimpleSnackbar-root-2",
    }
  }
  component={[Function]}
/>
`;
